### PR TITLE
llm finetuning

### LINKED ISSUE #13: Implement GPT-2 finetuning for sentiment generation
This consists of:

1. Loading the GPT-2 pretrained model.
2. Configuring the finetuning setup so that GPT-2 learns to complete prompts with a positive/negative review.

### PR DESCRIPTION
**Pull request by EdoardoPona**
issues #11 #12 #13

### CODE REVIEW

**Comment by EdoardoPona** [comment]
RL4LMs currently fails if the reward is not returned at each step. This was not an issue previously, we were only ever performing a single step.

**Comment by EdoardoPona** [comment]
https://github.com/allenai/RL4LMs/commit/e147dd3dee0e539cb96d97e3ffb851ef63c85fc0 implements a reward class that can be used at the same time by the metric object, without being instantiated
